apps/wallet/src/ui/app/pages/home/transfer-coin/index.tsx

```diff
@@ -172,67 +167,98 @@ function TransferCoinPage() {
         ({ suiObjects }) => suiObjects.loading && !suiObjects.lastSync
     );
 
+    const handleNextStep = useCallback(
+        (
+            formData: FormValues,
+            { setSubmitting }: FormikHelpers<FormValues>
+        ) => {
+            setCurrentStep((prev) => prev + 1);
+            setFormData(formData);
+            setSubmitting(false);
+        },
+        []
+    );
+
     if (!coinType) {
         return <Navigate to="/" replace={true} />;
     }
 
-    const StepOneForm = (
-        <Formik
-            initialValues={formData}
-            validateOnMount={true}
-            validationSchema={validationSchemaStepOne}
-            onSubmit={handleNextStep}
-        >
-            <StepOne
-                coinSymbol={coinSymbol}
-                coinType={coinType}
-                onClearSubmitError={handleOnClearSubmitError}
-                onAmountChanged={(anAmount) => setAmountToSend(anAmount)}
-            />
-        </Formik>
-    );
+    const StepOne = (
+        <>
+            <div className="mb-7">
+                <ActiveCoinsCard activeCoinType={coinType} />
+            </div>
 
-    const StepTwoForm = (
-        <Formik
-            initialValues={formData}
-            validateOnMount={true}
-            validationSchema={validationSchemaStepTwo}
-            onSubmit={onHandleSubmit}
-        >
-            <StepTwo
-                submitError={sendError}
-                coinSymbol={coinSymbol}
-                coinType={coinType}
-                gasBudgetEstimation={gasBudgetEstimation || null}
-                gasCostEstimation={gasBudgetEstimation || null}
-                gasEstimationLoading={isLoading}
-                onClearSubmitError={handleOnClearSubmitError}
-            />
-        </Formik>
+            <Formik
```

Comment:
Let's move this inside SendTokensForm and let it handle all required values. Then we can expose an onSubmit that will expose all necessary values for a transaction eg. amount, address, gasEstimation, isPaySuiAll (and in this case maybe include the Sui coin object ids). Also to allow initial values we can have input props like amount, address.

This way I think it will be clearer and this `TransferCoinPage` will only handle the transition between input and preview and finally submit